Naughts_and_Crosses.py

```diff
@@ -1,122 +1,237 @@
+        if who_goes_first == '1':
+            self.current_player = self.p1
+        elif who_goes_first == '2':
+            self.current_player = self.p2
         else:
-            print('I cannot do that Hal...')
-            print('Try again using only (y/n)')
-           start_game(self)
-
-def check_win(board_state):
-    if board_state in win_condition:
-        return True
-    else:
-        return False
-
-def check_draw():
-    if ' ' not in board_state:
-        print('Oh dear, looks like there\'s no more moves left!')
-        start_game(would_you_like_to_play)
-
-
-def printboard(list):
-    for l in list:
-        print('|'.join(l))
-
-def end_game():
-    if board_state in win_condition:
-        print('Well Done, you won! Good Job')
-        would_you_like_to_play = print('Would you like to play again?')
-
-
-
-class Player1:
-    def __init__(self,name, symbol):
-        self.name = input('What is your name?')
-        self.symbol = 'X'
-        # self.symbol = input('Which symbol would you like? (X/O)')
-
-    def choice_row():
-        x=()
-        while x not in range(3):
-            x= int(input('Pick what row you would like') - 1)
-            if x not in range(3):
-                print('That\'s not in the right format you douche, try again')
-        return x
-
-    def choice_column():
-        y=()
-        while y not in range(3):
-            y= int(input('Pick what column you would like') - 1)
-            if y not in range(3):
-                print('That\'s not in the right format you douche, try again')
-                y = ()
-        return y
-
-    def turn(self):
-        x_row = choice_row()
-        y_column = choice_column()
-        if board_state[x_row][y_column] == 'X' or board_state[x_row][y_column] == 'O':
-            print('Square\'s already full idiot') 
-            turn(self)
+            print("That's not a player number...")
+            self.first_turn()
+
+    def turn(self): #Checks the current player, makes them do a turn based on what type of player they are then
+        if self.current_player.player_type == 'H':  #takes the slot they used out of the empty spaces list
+            choices = self.current_player.human_choice()
+            if choices in self.board.empty_spaces:
+                self.board.board_state[choices[0]][choices[1]] = self.current_player.symbol
+                self.board.empty_spaces.remove(choices)
+
+            else:
+                print('That space is not available douche. Choose again')
+                self.turn()
         else:
-            board_state[x_row][y_column] = self.symbol
-            return board_state
-
-class AI():
-	def __init__(self, symbol):
-		self.symbol = 'O'
-    def turn(self):
-        x_row = randrange(3)
-        y_column = randrange(3)
-        if board_state[x_row][y_column] == 'X' or board_state[x_row][y_column] == 'O':
-            turn(self)
+            choices = self.current_player.ai_choice()
```

Comment:
Soooooo much repeated code.

Just get the choice, then do the filling/checking.